core/templates/services/rte-helper-modal.controller.ts

```diff
@@ -420,7 +432,8 @@ export class RteHelperModalComponent {
           customizationArgsDict[caName] = this.extractVideoIdFromVideoUrl(
             temp.toString()
           );
-        } else if (caName === 'text' && this.currentRteIsLinkEditor) {
+        } else if (caName === 'text' &&
+        this.componentId === this.COMPONENT_ID_LINK) {
```

Comment:
Have your top-level if/else here (the one started at line 344) be based on component type. Each if/else block should correspond to a single component type.

Reply:
@seanlip is dividing like this what you mean?
```ts
  save(): void {
    for (let index in this.customizationArgsForm.value) {
      this.tmpCustomizationArgs[index].value = (
        this.customizationArgsForm.value[index]);
    }
    this.externalRteSaveService.onExternalRteSave.emit();

    const customizationArgsDict: {
      [Prop in keyof CustomizationArgsForRteType]?:
      CustomizationArgsForRteType[Prop]
    } = {};
    // For the case of the math rich text components, we need to handle the
    // saving of the generated SVG file here because the process of saving
    // the SVG is asynchronous and the saving of SVG to the backend is to
    // be done only after the user clicks on the save button.
    // The saving of SVGs to the backend cannot be done in the math RTE editor
    // because the control is passed to this function as soon as the user
    // clicks on the save button.
    if (this.componentId === this.COMPONENT_ID_MATH) {
      // The tmpCustomizationArgs is guaranteed to have only one element for
      // the case of math rich text component.
      // We know that this is a math rich text component. Hence we can make the
      // the type more specific.
      const tmpCustomizationArgs = this.tmpCustomizationArgs as Extract<
        CustomizationArgsNameAndValueArray[number],
        { name: 'math_content' }
      >[];
      const svgFile = tmpCustomizationArgs[0].value.svgFile;
      const svgFileName = tmpCustomizationArgs[0].value.svg_filename;
      const rawLatex = tmpCustomizationArgs[0].value.raw_latex;
      if (rawLatex === '' || svgFileName === '') {
        this.alertsService.addWarning(
          'The rawLatex or svgFileName for a Math expression should not ' +
            'be empty.'
        );
        this.ngbActiveModal.dismiss('cancel');
        return;
      }
      const resampledFile =
        this.imageUploadHelperService.convertImageDataToImageFile(svgFile);

      let maxAllowedFileSize;
      if (
        this.contextService.getEntityType() ===
        AppConstants.ENTITY_TYPE.BLOG_POST
      ) {
        const ONE_MB_IN_BYTES = 1 * 1024 * 1024;
        maxAllowedFileSize = ONE_MB_IN_BYTES;
      } else {
        const HUNDRED_KB_IN_BYTES = 100 * 1024;
        maxAllowedFileSize = HUNDRED_KB_IN_BYTES;
      }
      if (resampledFile.size > maxAllowedFileSize) {
        this.alertsService.addInfoMessage(
          `The SVG file generated exceeds ${maxAllowedFileSize / 1024}` +
            ' KB. Please split the expression into smaller ones.' +
            "   Example: x^2 + y^2 + z^2 can be split as 'x^2 + y^2' " +
            "and '+ z^2'",
          5000
        );
        this.ngbActiveModal.dismiss('cancel');
        return;
      }
      if (
        this.contextService.getImageSaveDestination() ===
        AppConstants.IMAGE_SAVE_DESTINATION_LOCAL_STORAGE
      ) {
        this.imageLocalStorageService.saveImage(svgFileName, svgFile);
        const mathContentDict = {
          raw_latex: tmpCustomizationArgs[0].value.raw_latex,
          svg_filename: svgFileName,
        };
        const caName = tmpCustomizationArgs[0].name;
        customizationArgsDict[caName] = mathContentDict;
        this.ngbActiveModal.close(customizationArgsDict);
        return;
      }
      this.assetsBackendApiService
        .saveMathExpressionImage(
          resampledFile,
          svgFileName,
          this.contextService.getEntityType(),
          this.contextService.getEntityId()
        )
        .then(
          (response) => {
            const mathContentDict = {
              raw_latex: tmpCustomizationArgs[0].value.raw_latex,
              svg_filename: response.filename,
            };
            const caName = tmpCustomizationArgs[0].name;
            customizationArgsDict[caName] = mathContentDict;
            this.ngbActiveModal.close(customizationArgsDict);
          },
          (errorResponse) => {
            this.alertsService.addWarning(
              errorResponse.error || 'Error communicating with server.'
            );
            this.ngbActiveModal.dismiss('cancel');
          }
        );
    } else if (this.componentId === this.COMPONENT_ID_VIDEO) {
      for (let i = 0; i < this.tmpCustomizationArgs.length; i++) {
        const caName = this.tmpCustomizationArgs[i].name;
        if (caName === 'video_id') {
          const temp = this.tmpCustomizationArgs[i].value;
          customizationArgsDict[caName] = this.extractVideoIdFromVideoUrl(
            temp.toString()
          );
        }
      }
      this.ngbActiveModal.close(customizationArgsDict);
    } else if (this.componentId === this.COMPONENT_ID_LINK) {
      for (let i = 0; i < this.tmpCustomizationArgs.length; i++) {
        const caName = this.tmpCustomizationArgs[i].name;
        if (caName === 'text') {
        // Set the link `text` to the link `url` if the `text` is empty.
          (
          customizationArgsDict as {
            [Prop in CustomizationArgsNameAndValueArray[number]['name']]:
              CustomizationArgsNameAndValueArray[number]['value'];
          }
          )[caName] =
          this.tmpCustomizationArgs[i].value ||
          this.tmpCustomizationArgs[i - 1].value;
        }
      }
      this.ngbActiveModal.close(customizationArgsDict);
    } else {
      for (let i = 0; i < this.tmpCustomizationArgs.length; i++) {
        const caName = this.tmpCustomizationArgs[i].name;
        (
        customizationArgsDict as {
          [Prop in CustomizationArgsNameAndValueArray[number]['name']]:
            CustomizationArgsNameAndValueArray[number]['value'];
        }
        )[caName] = this.tmpCustomizationArgs[i].value;
      }
      this.ngbActiveModal.close(customizationArgsDict);
    }
    this.rteFieldValues.unsubscribe();
  }
```

Reply:
Yes, the top-level structure seems roughly correct. One question though, in the "else" clause, shouldn't the general logic there apply to all RTE components?

Also please apply [this comment](https://github.com/oppia/oppia/pull/19179#discussion_r1420866220) to the code, there are some places where it is a bit hard to tell where a new instruction starts.